src/coreclr/jit/compiler.cpp

```diff
@@ -4791,6 +4794,17 @@ void Compiler::compCompile(void** methodCodePtr, uint32_t* methodCodeSize, JitFl
                 DoPhase(this, PHASE_OPTIMIZE_VALNUM_CSES, &Compiler::optOptimizeCSEs);
             }
 
+            // Assertion prop can do arbitrary statement remorphing, which
+            // can clone code and disrupt our simpleminded SSA accounting.
```

Comment:
I believe all optimization phases (save copy propagation) can call morph an arbitrary statements.

Reply:
I think the ones upstream from AP are handled now. Assertion prop does this remorphing centrally and globally so it we would end up rescanning more stuff, and there's no expected downstream consumer (other than checking). Easy enough to try adding it later if we think there's value.

Reply:
I see. There is a sort of "SSA checker" in range check, would be nice to delete that, but I suppose it would require diverging between debug and release code (which seems undesirable for release-available data).